src/org/loklak/harvester/TwitterScraper.java

```diff
@@ -165,24 +208,39 @@ public static String prepareSearchURL(final String query) {
      */
     public static Timeline[] search(
             final BufferedReader br,
+            String filter,
             final Timeline.Order order,
             final boolean writeToIndex,
             final boolean writeToBackend) throws IOException {
+        Matcher match_video1;
+        Matcher match_video2;
+        Pattern[] video_url_patterns = {
+                Pattern.compile("youtu.be\\/[0-9A-z]+"),
+                Pattern.compile("youtube.com\\/watch?v=[0-9A-z]+")
+        };
         Timeline timelineReady = new Timeline(order);
         Timeline timelineWorking = new Timeline(order);
+        Set<String> filter_array = new HashSet<String>(Arrays.asList(
+                (filter.toLowerCase()).split(",")
+        ));
         String input;
         Map<String, prop> props = new HashMap<String, prop>();
         Set<String> images = new LinkedHashSet<>();
         Set<String> videos = new LinkedHashSet<>();
-        String place_id = "", place_name = "";
+        String place_id;

```

Comment:
this point about wrong images being shown is different issue related to #1135 i.e. storing wrong images.
filter=image just takes tweets which has images.